xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/renderer/xhtml/link/DocumentXHTMLLinkTypeRenderer.java

```diff
@@ -90,6 +102,37 @@ protected String computeLabel(ResourceReference reference)
         return this.linkLabelGenerator.generate(reference);
     }
 
+    private WantedLinkTitleGenerator getTitleGenerator(ResourceReference reference)
+    {
+        WantedLinkTitleGenerator titleGenerator = this.defaultTitleGenerator;
+        try {
+            titleGenerator = this.componentManager.getInstance(WantedLinkTitleGenerator.class,
+                reference.getType().getScheme());
+        } catch (Exception e) {
+            String message = "Could not find a [WantedLinkTitleGenerator] component to generate the wanted "
```

Comment:
you should use `WantedLinkTitleGenerator.class.getName()` instead of hardcoding it IMO.